### PR TITLE
Implement contains function for IP address

### PR DESCRIPTION
**Pull request by ebyhr**
Related to #12012

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/IpAddressFunctions.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+public final class IpAddressFunctions
+{
+    private IpAddressFunctions() {}
+
+    @Description("Determines whether given ip address exists in the CIDR")
+    @ScalarFunction("contains")
+    @SqlNullable
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean contains(@SqlType(StandardTypes.VARCHAR) Slice slice, @SqlType(StandardTypes.IPADDRESS) Slice value)
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/IpAddressFunctions.java]: Let's use more descriptive argument names like `cidr` and `address`

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/IpAddressFunctions.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        String format = slice.toStringUtf8();
+        CIDR cidr;
+        try {
+            cidr = CIDR.newCIDR(format);
+        }
+        catch (UnknownHostException e) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, String.format("Invalid CIDR format (%s)", format));
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/IpAddressFunctions.java]: For a single value, put it at the end:
```java
throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid CIDR value: " + format);

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/IpAddressFunctions.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.net.UnknownHostException;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+public final class IpAddressFunctions
+{
+    private IpAddressFunctions() {}
+
+    @Description("Determines whether given ip address exists in the CIDR")
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/IpAddressFunctions.java]: Capitalize IP

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/IpAddressFunctions.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    @ScalarFunction("contains")
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/IpAddressFunctions.java]: No need to give a function name when it matches the Java method name

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/IpAddressFunctions.java (hunk 1 above)
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/IpAddressFunctions.java]: This should return a primitive `boolean`

File: presto-docs/src/main/sphinx/functions/ipaddress.rst [hunk 5]
```diff
@@ -0,0 +1,13 @@
+====================
+IP address Functions
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/ipaddress.rst]: Capitalize Address

File: presto-docs/src/main/sphinx/functions/ipaddress.rst [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+====================
+
+
+.. function:: contains(format, address) -> boolean
+    :noindex:
+
+    Returns true if the ``address`` exists in the CIDR ``format``.::
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/ipaddress.rst]: Remove the period at the end, since the `::` will produce a colon

**Comment by electrum** [comment]
Overall this looks good and is a great PR: includes tests, documentation, etc.

**Comment by ebyhr** [comment]
@electrum Thank you for your review and kind help. Updated.

**Pull request closed without merging**